network/src/protocols/rpc/mod.rs

```diff
@@ -411,7 +412,8 @@ async fn handle_inbound_substream<TSubstream>(
                 );
             }
         }
-        notif => unreachable!(
+        notif => debug_checked_verify!(
```

Comment:
If you want to know how MIRAI can verify this, here is the story: The function is not public, so presumably we can find every call to this function. In order for this false verification to be unreachable, the callers should never supply an argument that is not NewInboundSubstream. When analyzing this function, that precondition is inferred. When analyzing a call site, the inferred precondition should be satisfied or there'll be a diagnostic message. Absent such messages, we can be reasonably confident that this statement will never be reached.

Reply:
Thanks for the explanations! 

I still haven't managed to run MIRAI on the network crate. It seems to take an absurd amount of time to compile (maybe because it is using a C++ library (GRPC)?)

I'll merge this PR once I manage to run it.

Reply:
Note that the network crate *does not* depend on gRPC